### PR TITLE
Add `disableSynchronization` option to vmr source mappings

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/4158

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs [hunk 1]
```diff
@@ -304,6 +309,11 @@ private async Task<bool> UpdateRepositoryRecursively(
 
         foreach (VmrDependencyUpdate update in updates)
         {
+            if (update.Mapping.DisableSynchronization)
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: Technically a similar `if` should go into `VmrInitializer`. Just for completness

File: test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs [hunk 2]
```diff
@@ -91,6 +102,10 @@ public async Task RecursiveUpdatePreservesDependencyVersionTest()
             Constants.DependencyTemplate,
             Constants.SecondRepoName, SecondRepoPath, newSecondRepoSha);
 
+        var syncDisabledRepoDependency = string.Format(
```
Comment (premun) [inline test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs:105]: Should this not be added into line 109?

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: I would not add tests for that for the VmrInitializer though

**Review by premun — COMMENTED** [review]
@dkurepa we also warn when there's a mapping that did not show up in the `Version.Details.xml` dependency tree:
https://github.com/dkurepa/arcade-services/blob/3b465890dc9dfe3aadd77c9ba640f3348149b6dd/src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs#L286-L295

I think we might want to not warn about disabled mappings

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs (hunk 1 above)
Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: Ah right, will add it. Also will need to modify the test so the repo sync gets disabled after its initialized

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: Ah, in that case let's not do that

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: It's call sync disabled, not initialization disabled, so let's just not do it

Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:312]: that was kind of my original thought, that the repo should still get initialized

File: test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs (hunk 2 above)
Comment (dkurepa) [reply test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRecursiveSyncTests.cs:105]: ah right, forgot to do that, thanks

**Comment by dkurepa** [comment]
> @dkurepa we also warn when there's a mapping that did not show up in the `Version.Details.xml` dependency tree: https://github.com/dkurepa/arcade-services/blob/3b465890dc9dfe3aadd77c9ba640f3348149b6dd/src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs#L286-L295
> 
> I think we might want to not warn about disabled mappings

yeah, true, I added the condition to the where clause to not add disabled mappings, thanks

**Review by premun — APPROVED** [review]

**Pull request merged**